test/test_helper.rb

```diff
@@ -100,6 +110,17 @@ def user(item)
     users[item] = true
   end
 
+  def topic(name)
+    return topics[name] if topics.key?(name)
+    topics[name] = client.search_repositories("topic:#{name}").total_count > 0
+    @@topic_request_count += 1
+    sleep client.rate_limit.resets_in if client.rate_limit.remaining == 0
+    topics[name]
+  rescue Octokit::TooManyRequests
+    topics[:skip_requests] = true
+    topics[name] = true
+  end
+
```

Comment:
I love the idea here. However, I'm not sure that we want to `sleep` until the rate limit resets. I am not sure how long that cycle usually lasts, but we'll have to pay that price on all PRs after a rate limit is hit.

What if instead of sleeping, we set the `skip_requests` value when we hit `0`?

```suggestion
  def topic(name)
    return topics[name] if topics.key?(name)

    if client.rate_limit.remaining > 0
      @@topic_request_count += 1
      topics[name] = client.search_repositories("topic:#{name}").total_count > 0
    else
      topics[:skip_requests] = true
      topics[name] = true
    end
    
    topics[name]
  rescue Octokit::TooManyRequests
    topics[:skip_requests] = true
    topics[name] = true
  end

```

The rescue might be redundant in this case, but I'm leaving it in there to match what we're doing in other places. Although, I like this pattern better. If you had the time to modify the rest of the methods to match this, and if it works, I'd be +1 for it.

Reply:
As I understand it, what `skip_requests` does, is to only check topics until the rate limit is reached and leave the rest unchecked, so we don't have to wait? Doesn't sound very trustworthy to me, to be honest. However, this might be solved by a GraphQL implementation&mdash;I definitely look into that!

Reply:
Yeah, I wrote all this a very long time ago and I think I was doing it so that we could simply get passed the failing checks, then I swapped over to the graphql stuff and all of this was no longer a problem.

Reply:
It's definitely possible via GraphQL. I'm implementing it right now. Stay tuned!

Reply:
![grafik](https://github.com/github/explore/assets/11225821/6c46e125-1fd1-4dd4-9973-bbda120a07cb)
My assumption is, that 1289 topics might be too much for the API—can that be the reason? If it's not, the error message isn't hinting at anything specifically ^^

Reply:
I was wrong. I did a little fiddling, and it turns out that the API doesn't expect the topic's `repositories` property to be accessed if it's not curated. Is there a way around?

Reply:
I think you can access the `totalCount` on the `repositories` property for topics that don't have any repos still:

```graphql
query {
  topic(name:"kart-racing") {
    repositories{
      totalCount
    }
    relatedTopics {
      id
    }
  }
}
```

result:

```graphql
{
  "data": {
    "topic": {
      "repositories": {
        "totalCount": 0
      },
      "relatedTopics": []
    }
  }
}
```

Reply:
Ah, you are right. My query was:
```graphql
query {
  config: topic(name:"config") {
    repositories{
      totalCount
    }
    relatedTopics {
      id
    }
  }
}
```

But it's not that 5,819,971 repositories might be too much, is it?